### ISSUE TITLE
test: object store test starts its own node, ignoring the node provided by test framework

### ISSUE THREAD

**Issue by kbr-scylla** [scylladb/scylladb#16268]
This is pretty confusing behavior: let's say you run the test with test.py (is there another way to run it?). The test.py framework creates a single-node cluster and gives it to the test to use, but the test ignores it; instead, it creates its own nodes.
```python3
async def test_basic(test_tempdir, s3_server, ssl):
    '''verify ownership table is updated, and tables written to S3 can be read after scylla restarts'''

    with managed_cluster(test_tempdir, ssl, s3_server) as cluster:
```

So we're dealing with multiple nodes / clusters, it's not clear which logs should be looked at when investigating a failure.

If the test wants to start a cluster with custom config, perhaps it would be more appropriate to turn it into a topology-type test in suite that provides 0-node clusters? (like topology_custom)

cc @xemul @tchaikov @nyh

**Comment by xemul** [comment]
> If the test wants to start a cluster with custom config,

It wants test with custom config, it also wants to restart cluster and wants to kick cluster with HUP to make it re-read config. 

It was faster and simpler to write home-brew `managed_cluster` thing rather than fixing existing pylib's `Cluster` to do we we needed :man_shrugging: . If I understood @nyh correctly, similarly it was one of the reasons why we ended up having yet another "cluster management" in scylla unit tests, rather than re-using one from, say, dtests or ccm

**Comment by xemul** [comment]
>  it was one of the reasons why we ended up having yet another "cluster management" in scylla unit tests

here: https://github.com/scylladb/scylladb/issues/16006#issuecomment-1828234308

**Comment by kbr-scylla** [comment]
> It wants test with custom config, it also wants to restart cluster and wants to kick cluster with HUP to make it re-read config.

> It was faster and simpler to write home-brew managed_cluster thing rather than fixing existing pylib's Cluster to do we we needed 🤷‍♂️

Custom config: `ManagerClient.server_add` has `config: Optional[dict[str, Any]] = None` parameter
restart cluster: `ManagerClient.server_restart` (for each node)
kick with HUP: `ManagerClient.server_update_config` does exactly that if  the node is running

**Comment by nyh** [comment]
> If the test wants to start a cluster with custom config, perhaps it would be more appropriate to turn it into a topology-type test in suite that provides 0-node clusters? (like topology_custom)

Right. This is why I ended up creating new tests in the (irrelevant named) `test/topology_experimental_raft/` - because this directory doesn't create its own cluster and lets each test create whatever cluster it chooses.

The fix for this issue, if it's at all an issue, is trivial - in test/object_store/suite.yaml put
```
cluster:
  initial_size: 0
```
Like in test/topology_experimental_raft/suite.yaml. Or just move this test into the existing topology_experimental_raft directory.
